src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Interface/IKeyVault.cs

```diff
@@ -0,0 +1,40 @@
+using Microsoft.Azure.KeyVault;
+using Microsoft.Azure.KeyVault.WebKey;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+
+namespace Altinn.App.Services.Interface
+{
+    /// <summary>
+    /// Interface for Key Vault service
+    /// </summary>
+    public interface IKeyVault
+    {
+        /// <summary>
+        /// Gets a key from key vault.
+        /// </summary>
+        /// <param name="keyId">The id of the key.</param>
+        /// <returns>The key as a JSON web key.</returns>
+        Task<JsonWebKey> GetKeyAsync(string keyId);
+
+        /// <summary>
+        /// Gets a secret from key vault.
+        /// </summary>
+        /// <param name="secretId">The id of the secret.</param>
+        /// <returns>The secret value.</returns>
+        Task<string> GetSecretAsync(string secretId);
+
+        /// <summary>
+        /// Gets the latest version of a certificate from key vault.
+        /// </summary>
+        /// <param name="certificateId">The id of the certificate.</param>
+        /// <returns>The certificate as a byte array.</returns>
+        Task<byte[]> GetCertificateAsync(string certificateId);
+
+        /// <summary>
+        /// Gets the key vault client.
+        /// </summary>
+        /// <returns>The key vault client.</returns>
+        KeyVaultClient GetKeyVaultClient();
```

Comment:
By exposing the client I figure we don't need to abstract the versioning of the certificates. They can solve that themselves.

Reply:
For most app developers they simply need to retrieve the lastest certificate and aren't bothered with which versions are available in key vault. So in order not to make the standard method complex I figured those who will need to use different versions of the same certificate in different settings could implement that themselves? Ref acceptance criteria: _Client gives access to all versions of a specific certificate. Latest first._